src/parcsr_ls/par_gsmg.c

```diff
@@ -520,7 +520,7 @@ hypre_BoomerAMGCreateSmoothVecs(void         *data,
 
       for (i = 0; i < num_sweeps; i++)
       {
-         if (smooth_option == 6)
+         if ((smooth_num_levels > level) && (smooth_option == 6))
```

Comment:
Was this a bug?

Reply:
Yes! I should have marked these bugs...

Here, we should run the complex smoother only in the requested levels, hence the check `(smooth_num_levels > level)`